### PR TITLE
Add 'compile' command as an alias of the 'build' command.

### LINKED ISSUE #502: Provide `compile` as a replacement/alias for the `build` command
The help message uses "compile" ("compile program file") and most compilers typically have a `c` suffix or use `-c` or `--compile`.

### PR DESCRIPTION
**Pull request by keplersj**
Fixes #502

### CODE REVIEW

File: src/compiler/crystal/command.cr [hunk 1]
```diff
@@ -8,6 +8,7 @@ class Crystal::Command
     Command:
         init                     generate a new project
         build                    compile program
```
Comment (jhass) [inline src/compiler/crystal/command.cr]: Actually perhaps we should even already remove it here.

File: src/compiler/crystal/command.cr [hunk 2]
```diff
@@ -55,6 +56,9 @@ class Crystal::Command
       when "build".starts_with?(command)
         options.shift
         build
+      when "compile".starts_with?(command)
```
Comment (jhass) [inline src/compiler/crystal/command.cr]: A when branches takes a comma separated list of tests ;)

File: src/compiler/crystal/command.cr [hunk 3]
```diff
@@ -155,6 +159,11 @@ class Crystal::Command
     config.compile
   end
 
+  private def compile
```
Comment (asterite) [inline src/compiler/crystal/command.cr]: I would rename the current `build` with `compile` instead

File: src/compiler/crystal/command.cr [hunk 4]
```diff
@@ -52,9 +52,9 @@ class Crystal::Command
       when "init".starts_with?(command)
         options.shift
         init
-      when "build".starts_with?(command)
+      when "build".starts_with?(command), "compile".starts_with?(command)
         options.shift
-        build
+        compile command
```
Comment (jhass) [inline src/compiler/crystal/command.cr]: I think the usage saying "compile" when "build" is still used would actually be a good thing :)

**Comment by jhass** [comment]
Great, now some squashing would be great here too ;)

**Comment by keplersj** [comment]
Squashed!

**Comment by ysbaddaden** [comment]
I thought we didn't like aliases?

**Comment by jhass** [comment]
We don't, but then this is so essential that we need to phase it out IMO. So the current plan is to add this as an alias, update all references we can update to compile and then at some later release remove build.

**Comment by jhass** [comment]
Also our own `Makefile` would break otherwise.

**Comment by ysbaddaden** [comment]
Thanks! There was no indication about a deprecation of the build command.

**Comment by asterite** [comment]
Oh, a deprecation warning is very good indeed, so users can learn about the new command.

**Comment by keplersj** [comment]
Travis is green!

**Comment by jhass** [comment]
Thank you!

**Comment by keplersj** [comment]
Woot!

**Pull request merged**